File: lib/features/properties/domain/entities/property_entity.dart
enum OperationType { sale, rent }

enum PropertyStatus { available, reserved, sold, rented, inactive }

class PropertyEntity {
  final String id;
  final String ownerId;
  final String title;
  final String description;
  final double price;
  final OperationType operationType;
  final PropertyStatus status;
  final String city;
  final String zone;
  final String? colony;
  final int bedrooms;
  final int bathrooms;
  final bool hasGarage;
  final bool hasGarden;
  final double area;
  final List<String> photos;
  final DateTime createdAt;

  const PropertyEntity({
    required this.id,
    required this.ownerId,
    required this.title,
    required this.description,
    required this.price,
    required this.operationType,
    required this.status,
    required this.city,
    required this.zone,
    this.colony,
    required this.bedrooms,
    required this.bathrooms,
    required this.hasGarage,
    required this.hasGarden,
    required this.area,
    required this.photos,
    required this.createdAt,
  });
}

File: backend/app/core/domain/entities/property_entity.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import List, Optional
from enum import Enum

class OperationType(str, Enum):
    sale = "sale"
    rent = "rent"

class PropertyStatus(str, Enum):
    available = "available"
    reserved = "reserved"
    sold = "sold"
    rented = "rented"
    inactive = "inactive"

@dataclass
class PropertyEntity:
    id: str
    owner_id: str
    title: str
    description: str
    price: float
    operation_type: OperationType
    city: str
    zone: str
    area: float
    status: PropertyStatus = PropertyStatus.available
    colony: Optional[str] = None
    bedrooms: int = 1
    bathrooms: int = 1
    has_garage: bool = False
    has_garden: bool = False
    photos: List[str] = field(default_factory=list)
    cluster: Optional[int] = None
    created_at: Optional[datetime] = None
